src/pages/api/sepana/utils.ts

```diff
@@ -0,0 +1,250 @@
+import axios from 'axios'
+import {
+  SepanaProject,
+  SepanaProjectJson,
+  SepanaSearchResponse,
+} from 'models/sepana'
+
+const headers = (type: 'read' | 'read/write' | 'admin') => {
+  let key: string
+
+  switch (type) {
+    case 'read':
+      if (!process.env.SEPANA_READ_API_KEY) {
+        throw new Error('Missing SEPANA_READ_API_KEY')
+      }
+      key = process.env.SEPANA_READ_API_KEY
+      break
+    case 'read/write':
+      if (!process.env.SEPANA_READ_WRITE_API_KEY) {
+        throw new Error('Missing SEPANA_READ_WRITE_API_KEY')
+      }
+      key = process.env.SEPANA_READ_WRITE_API_KEY
+      break
+    case 'admin':
+      if (!process.env.SEPANA_ADMIN_API_KEY) {
+        throw new Error('Missing SEPANA_ADMIN_API_KEY')
+      }
+      key = process.env.SEPANA_ADMIN_API_KEY
+      break
+  }
+
+  return {
+    'x-api-key': key,
+    'Content-Type': 'application/json',
+  }
+}
+
+/**
+ * Exhaustively queries all Sepana records.
+ *
+ * @returns Promise containing all project docs from Sepana database
+ */
+export async function queryAllSepanaProjects() {
+  return axios.post<SepanaSearchResponse<SepanaProject>>(
+    `${process.env.NEXT_PUBLIC_SEPANA_API_URL}/search`,
+    {
+      engine_ids: [process.env.SEPANA_ENGINE_ID],
+      query: {
+        match_all: {},
+      },
+      size: 10000, // TODO: Update this before we hit 10k projects
+      page: 0,
+    },
+    {
+      headers: headers('read'),
+    },
+  )
+}
+
+/**
+ * Search all Sepana project records, using preconfigured elastic search field weights and script source.
+ *
+ * @param query text to search
+ * @param pageSize max number of results to return
+ * @returns Promise containing project docs from Sepana database matching search params. If no query text is supplied, returns all projects
+ */
+export async function searchSepanaProjects(query = '', pageSize?: number) {
```

Comment:
this is where sepana "default search logic" is defined. note that we're including matches to name, handle, and description, but weighting name + handle higher. we're also prioritizing results from those matches that have a higher volume or trending score.